virtool/labels/api.py

```diff
@@ -1,180 +1,160 @@
+from typing import List, Union
+
 from aiohttp.web_exceptions import HTTPBadRequest, HTTPNoContent
+from aiohttp_pydantic import PydanticView
+from aiohttp_pydantic.oas.typing import r200, r201, r204, r400, r404
 from sqlalchemy import select
 from sqlalchemy.exc import IntegrityError
 from sqlalchemy.ext.asyncio import AsyncSession
 
 import virtool.http.routes
-import virtool.validators
 from virtool.api.response import EmptyRequest, NotFound, json_response
 from virtool.mongo.transforms import apply_transforms
-from virtool.http.schema import schema
 from virtool.labels.db import SampleCountTransform
 from virtool.labels.models import Label
 from virtool.pg.utils import get_generic
-
+from virtool.labels.oas import CreateNewLabelSchema, EditExistingLabelSchema
+from virtool.data_model.label import VirtoolLabel
 routes = virtool.http.routes.Routes()
 
 
-@routes.get("/labels")
-async def find(req):
-    """
-    Get a list of all label documents in the database.
+@routes.view("/labels")
+class LabelsView(PydanticView):
 
-    """
-    term = req.query.get("find")
+    async def get(self) -> Union[r200[List[VirtoolLabel]], r400]:
```

Comment:
Response models should not be prefixed:
```suggestion
    async def get(self) -> Union[r200[List[LabelMinimal]], r400]:
```

We want the core models to be importable elsewhere so they can be used to model what clients will get from the API. Please change elsewhere as well.

Lists of resources will also always use the minimal representation.